src/coreclr/tools/Common/TypeSystem/IL/EcmaMethodIL.cs

```diff
@@ -11,7 +11,10 @@
 
 namespace Internal.IL
 {
-    public sealed partial class EcmaMethodIL : MethodIL
+    // Marker interface implemented by EcmaMethodIL and EcmaMethodILScope
+    public interface IEcmaMethodIL { }
```

Comment:
Nit: might be faster to just do the two `is` checks instead of testing for the interface. Both EcmaMethodIL and EcmaMethodILScope are sealed so an `is` is quick.